### DISCUSSION TITLE
How about adding support for image captions?

### DISCUSSION THREAD

**Discussion by vetbactstaff** [datenstrom/community#353] [Ask a question] [answered]
The Gallery plugin looks very good, but I can't find any information about how to include captions. I'd like one short caption to go with each image and for it to be visible in the popup mode.

Since the plugin uses PhotoSwipe v4.1.2, I've looked at http://photoswipe.com/ and the software seems to be capable of displaying captions, but can it be done when used in Yellow? If so, where do you put the captions and how do you sync them with the images?

**Comment by markseuffert** [comment, marked as answer]
I added captions to the [gallery extension](https://github.com/annaesvensson/yellow-gallery). I wasn't sure where's a good place to put the captions, for now they can be defined in the [language settings](https://datenstrom.se/yellow/help/how-to-change-the-system#language-settings). Here's an example:

```
photo-1314380.jpg: This is an example caption.
photo-1425315.jpg: Sunset at the beach. Time to make a photo.
photo-2387365-fika-time.jpg: Fika is an important part of life in Sweden.
```

What do you think? Does this work for you?

**Comment by vetbactstaff** [comment]
Thank you very much for putting that in! Unfortunately, I have not been able to make it work yet. I have deleted the 5 files in system/plugins/gallery*, downloaded and installed the new version so that I now have files timestamped "aug 14 00:44". I have added a few rows with image names and captions to system/config/text.ini, but the captions don't show up, even after I flushed the browser cache and restarted the web server. Is there anything else I should do to make it work?

I think that putting the captions in system/config/text.ini would work for me, but in general it might be useful to have the captions more closely linked to the images. Perhaps they could go in a textfile called "captions.txt" in the same directory as the images themselves, e.g. media/images/album-fika/captions.txt? An advantage would be that it would be easier to remember to delete the captions when an album is removed. It would also be easier to have different language versions of the captions without having to rename the image files.

**Comment by markseuffert** [comment]
If the captions don't show up on your website, first I would check that the filenames are correct. Then I would download the developer kit, add the example captions (to the English language section), see if everything works as expected.

Hope this helps.

**Comment by vetbactstaff** [comment]
I have now downloaded a new version of the developer kit, made 'www-data' the owner of the directory ('chown -R www-data:www-data ./yellow-developers-master) and added the following lines to the end of system/config/text.ini:
```
photo-1314380.jpg: This is an example caption.
photo-1425315.jpg: Sunset at the beach. Time to make a photo.
photo-2387365-fika-time.jpg: Fika is an important part of life in Sweden.
```
then added this line to the first page of the demo:

```
[gallery photo-2387365-fika-time.jpg]
```
The image shows up as a single image gallery, but unfortunately I still don't see any captions...

**Comment by markseuffert** [comment]
The developer kit is multi language. Did you add the captions to the English language section?

**Comment by vetbactstaff** [comment]
No, I did not. I just put them at the end of the file. Now that I have moved the captions to the proper language sections it works. Thank you very much for adding this functionality! And for explaining how to use it.

**Comment by markseuffert** [comment]
I'll update the developer kit to make it easier to get started. Thank you for testing the gallery. 😄
